exec/java-exec/src/main/java/org/apache/drill/exec/work/metadata/ServerMetaProvider.java

```diff
@@ -76,7 +76,7 @@
       .setReadOnly(false)
       .setGroupBySupport(GroupBySupport.GB_UNRELATED)
       .setLikeEscapeClauseSupported(true)
-      .setNullCollation(NullCollation.NC_AT_END)
+      .setNullCollation(NullCollation.NC_HIGH)
```

Comment:
When sorting in Drill, the detailed sort spec is set in the {{ExternalSort}} operator definition. This thing is a bit complex. One can control sort order (ASC, DESC) and nulls position (LOW, HIGH, UNSPECIFIED.)

Data sorts according to ASC, DESC.
Nulls sort as follows:

HIGH: last if ASC, first if DESC
LOW: first if ASC, last if DESC
UNSPECIFIED: always high

If the planner has no way of setting the nulls ordering from a SQL query, then the value is UNSPECIFIED, which means nulls always sort last as Jinfeng explained.